scripts/fix-comments.py

```diff
@@ -0,0 +1,139 @@
+    # (to match qualified names)
+    s = s.replace('.', 'Ᾰ')
+    # re.DOTALL means that `.` can also match a newline.
+    # `\A` and `\Z` match only at the start/end of the string respectively.
+    w = re.findall(r'(?:\b|\A).+?(?:\b|\Z)', s, flags=re.DOTALL)
+    for i in range(len(w)):
+        w[i] = w[i].replace('Ᾰ', '.')
+        w[i] = name_map.get(w[i], w[i])
+    return ''.join(w)
+
+def process_backticked_names(s):
+    w = s.split(sep='`')
+    for i in range(len(w)):
+        if i % 2 == 1:
+            w[i] = replace_names(w[i])
+    return '`'.join(w)
+
+rewritten_contents = ''
+
+in_block_comment = False
+in_line_comment = False
+prev_char = None
+comment_so_far = None           # contains end marker but not begin marker
+
+def finish_comment():
+    global rewritten_contents
+    global in_block_comment
+    global in_line_comment
+    global comment_so_far
+    if comment_so_far is not None:
+        rewritten_contents += process_backticked_names(comment_so_far)
+        in_block_comment = False
+        in_line_comment = False
+        comment_so_far = None
+
+with open(leanfile) as F:
+    while 1:
+        char = F.read(1)
+        if not char:
+            finish_comment()
+            break
+
+        if in_block_comment or in_line_comment:
+            comment_so_far = comment_so_far + char
+        else:
+            rewritten_contents += char
+
+        if in_block_comment and prev_char == '-' and char == '/':
+            finish_comment()
+
+        if in_line_comment and char == '\n':
+            finish_comment()
+
+        if comment_so_far is None and prev_char == '/' and char == '-':
+            in_block_comment = True
+            comment_so_far = ''
+
+        if comment_so_far is None and prev_char == '-' and char == '-':
+            in_line_comment = True
+            comment_so_far = ''
+
+        prev_char = char
+
+def mktree(reversed_path_list, sha, tree=True):
+    if reversed_path_list == []:
+        return sha
+    hd, *tl = reversed_path_list
+    if tree:
+        inp = f"040000 tree {sha}\t{hd}"
+    else:
+        inp = f"100644 blob {sha}\t{hd}"
+    tree_sha = subprocess.run(
+        ['git', 'mktree'],
+        cwd=root_dir,
+        input=inp.encode('utf-8'),
+        capture_output=True).stdout.decode().rstrip()
+    return mktree(tl, tree_sha)
+
+path_list = subprocess.run(
+    ['git', 'ls-files', '--full-name', leanfile],
+    capture_output=True).stdout.decode().rstrip().split(sep='/')
+
+blob_sha = subprocess.run(
+    ['git', 'hash-object', '-w', '--stdin'],
+    input=rewritten_contents.encode('utf-8'),
+    cwd=root_dir,
+    capture_output=True).stdout.decode().rstrip()
+
+tree_sha = mktree(reversed(path_list), blob_sha, tree=False)
+
+subprocess.run(['git', 'restore', '--patch', '--source=' + tree_sha, '--', leanfile])
+
+r = subprocess.run(['git', 'diff', '--quiet', leanfile])
+if r.returncode != 0:           # file was changed
```

Comment:
```suggestion
if r.returncode == 1:           # file was changed
```
since if `git diff` fails it will produce some other exit code I think